okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -253,9 +255,10 @@ public OkHttpClientStream newStream(MethodDescriptor<?, ?> method,
         Headers.createRequestHeaders(headers, defaultPath, defaultAuthority);
 
     SettableFuture<Void> pendingFuture = null;
+    boolean shouldGoAway = false;
     synchronized (lock) {
```

Comment:
Specifically, `stream.allocated()`... However, it looks like that AbstractStream itself is simply not-threadsafe when calling into the listener as well. This may need a bigger discussion with @nmittler.